src/Board.cpp

```diff
@@ -17,6 +17,17 @@ Board::Board(int rowNumber,
     board_.assign(rowNumber_, singleRow);
 }
 
+Board::Board(std::string filename)
+    : rowNumber_(JSONFileReader(filename).readRowsNumber()),
```

Comment:
You can use delegating constructors from C++14 and call the other constructor here on an initilizing list.